crates/sui-core/src/authority.rs

```diff
@@ -1215,6 +1215,11 @@ impl AuthorityState {
                 transaction_dependencies,
             )
         } else {
+            // insert transaction digests that deleted a shared object that this transaction had
+            // as input but no longer exists due to sequencing order into transaction dependencies
+            for (_, digest) in deleted_shared_objects {
+                transaction_dependencies.insert(digest);
```

Comment:
If Tx1 deletes a shared object, Tx2 uses it, and then Tx3 uses it again. Would this change make Tx1 a dependency of Tx3 or Tx2 a dependency of Tx3? I think Tx2 should be a dependency of Tx3, because Tx3 must execute after Tx2, not Tx1.

Reply:
Each effects has this field `shared_objects`, which commits to the version of each input shared object, including already-deleted shared objects. So in that sense they must follow sequential execution order.

Reply:
But that seems to imply that we would commit to a non-existent shared object version in the effects?

Reply:
If both tx2 and tx3 will fail because tx1 already executed a deletion, then they both now depend on tx1 in addition to any existing dependencies. If the cert tx2 fails due to cert denied, in the current implementation the other objects in the transaction do not have their versions incremented, and thus no mutations are committed on those objects. Because of this, transaction execution in either order tx1 -> tx2 -> tx3 or tx1 -> tx3 -> tx2 will result in the same effects, thus I don't think tx3 needs to also depend on tx2. Do you disagree with this?